src/compiler/scala/tools/nsc/backend/jvm/opt/InlineInfoAttribute.scala

```diff
@@ -67,13 +67,10 @@ case class InlineInfoAttribute(inlineInfo: InlineInfo) extends Attribute(InlineI
     }
 
     // The method count fits in a short (the methods_count in a classfile is also a short)
-    result.putShort(inlineInfo.methodInfos.size)
+    result.putShort(inlineInfo.methodInfosSorted.size)
 
     // Sort the methodInfos for stability of classfiles
-    for ((nameAndType, info) <- inlineInfo.methodInfos.toList.sortBy(_._1)) {
-      val (name, desc) = nameAndType.span(_ != '(')
-      // Name and desc are added separately because a NameAndType entry also stores them separately.
-      // This makes sure that we use the existing constant pool entries for the method.
+    for (((name, desc), info) <- inlineInfo.methodInfosSorted) {
       result.putShort(cw.newUTF8(name))
       result.putShort(cw.newUTF8(desc))
```

Comment:
The second time that ASM calls this method, these lookups are fast because the `name`/`string` are the same s `String` instances that are interned in its constant pool from from the first pass, so `String.equals` need not compare the string contents.

It would also be possible to save the constant pool references in fields of `info`, but that's pretty ugly and doesn't save much.